src/main/java/se/sundsvall/partyassets/service/mapper/StatusMapper.java

```diff
@@ -0,0 +1,45 @@
+package se.sundsvall.partyassets.service.mapper;
+
+import static java.util.Collections.emptyList;
+import static java.util.Optional.ofNullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+
+import se.sundsvall.partyassets.api.model.Status;
+import se.sundsvall.partyassets.integration.db.model.StatusEntity;
+
+public class StatusMapper {
+
+	private StatusMapper() {}
+
+	public static Map<Status, List<String>> toReasons(final List<StatusEntity> entities) {
+		return ofNullable(entities).orElse(emptyList()).stream()
+			.map(entity -> Map.entry(Status.valueOf(entity.getName()), toReasons(entity)))
+			.collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+	}
+
+	public static List<String> toReasons(final StatusEntity entity) {
```

Comment:
Is this method needed? 

Is there really any drawback to doing `entity.getReasons()`; straight away?

Reply:
The method can be questioned, but I thought that it was nice to have all mapping grouped together at one place. But, I can agree with you that it just adds an extra unnecessary layer. Will remove.